api_docs.md

```diff
@@ -0,0 +1,58 @@
+
```

Comment:
Hooray for autogenerated docs! There are lots of ways to clean this up later and make it more readable for someone using the api, but this looks great!